core/src/main/scala/cats/MonadError.scala

```diff
@@ -77,6 +77,33 @@ trait MonadError[F[_], E] extends ApplicativeError[F, E] with Monad[F] {
   def redeemWith[A, B](fa: F[A])(recover: E => F[B], bind: A => F[B]): F[B] =
     flatMap(attempt(fa))(_.fold(recover, bind))
 
+  /**
+   * Reifies the value or error of the source and performs an effect on the result,
+   * then recovers the original value or error back into `F`.
+   *
+   * Note that if the effect returned by `f` fails, the resulting effect will fail too.
+   *
+   * Alias for `fa.attempt.flatTap(f).rethrow` for convenience.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> import scala.util.{Try, Success, Failure}
+   *
+   * scala> def checkError(result: Either[Throwable, Int]): Try[String] = result.fold(_ => Failure(new java.lang.Exception), _ => Success("success"))
```

Comment:
just a note, but don't consider it a blocker for the PR. I personally like the example in your comment [here](https://github.com/typelevel/cats/pull/3459#discussion_r437766077)
 
```scala
fa.attemptTap {
  case Right(a) => log.info(s"succeeded with $a")
  case Left(e) => log.error(s"failed with $e")
}
``` 
It describes very well a real use case for `attemptTap`. A simple program could be simulated with `State`. Something like
```scala
/**
 * {{{
 * scala> import cats._, data._, implicits._
 * scala> type F[A] = EitherT[State[String, *], String, A]
 * scala> object log { def info(m: String): F[Unit] = EitherT.liftF(State.modify(_ => s"info: $m")); def error(m: String): F[Unit] = EitherT.liftF(State.modify(_ => s"error: $m")) }
 *
 * scala> val prog1: F[Int] = EitherT.liftF(State.pure(1))
 * scala> val prog2: F[Int] = "this one failed".raiseError[F, Int]
 *
 * scala> val run1 = prog1.attemptTap { case Right(a) => log.info(a.toString); case Left(e) => log.error(e) }
 * scala> val run2 = prog2.attemptTap { case Right(a) => log.info(a.toString); case Left(e) => log.error(e) }
 *
 * scala> run1.value.runS("init").value
 * res0: String = info: 1
 *
 * scala> run2.value.runS("init").value
 * res1: String = error: this one failed
 * }}}
 */
```